src/common/cluster/cluster.ts

```diff
@@ -425,10 +429,20 @@ export class Cluster implements ClusterModel, ClusterState {
    * @internal
    */
   private async refreshAccessibility(): Promise<void> {
-    this.isAdmin = await this.isClusterAdmin();
-    this.isGlobalWatchEnabled = await this.canUseWatchApi({ resource: "*" });
-    this.allowedNamespaces = await this.getAllowedNamespaces();
-    this.allowedResources = await this.getAllowedResources();
+    const proxyConfig = await this.getProxyKubeconfig();
+    const canI = this.dependencies.createAuthorizationReview(proxyConfig);
```

Comment:
Is this where the optimization is?

Reply:
Yes, it creates the API object only once instead of dozens or hundred of times.

Reply:
I can see here where it creates it once instead of twice like the old code. Where else is the creation avoided?

Reply:
The big win is from `getAllowedResources` which calls `canI` in a loop

Reply:
ah, right, the class member version of `canI()` was creating the api each time it was called.

Reply:
Exactly